### PR TITLE
fix: comparision operations between incompatible operands

### LINKED ISSUE #34139: [Bug]: milvus crashes by illegal search expression
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.4.4
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus latest
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Create a collection with an int64 field named "xxx", insert some data, search with illegal expression "\"xxx\" > 500 && \"xxx\" < 1000", Milvus crashes.

### Expected Behavior
Milvus should return an error to client instead of crashes.

_No response_

### Steps To Reproduce

```markdown
import random

from pymilvus import (
    MilvusClient,
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)

connections.connect(host='localhost', port='19530')
print(utility.get_server_version())

collection_name = "AAA"
dim = 512
metric_type = "L2"


fields=[
    FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=False),
    FieldSchema(name="int64_field", dtype=DataType.INT64),
    FieldSchema(name="vector", dtype = DataType.FLOAT_VECTOR, dim=dim),
]

schema = CollectionSchema(fields=fields, enable_dynamic_field=True)

if utility.has_collection(collection_name):
    utility.drop_collection(collection_name)

collection = Collection(name=collection_name, schema=schema)
print(f"Collection '{collection_name}' created")

count = 10000
data = [
    [i for i in range(count)],
    [i for i in range(count)],
    [[random.random() for _ in range(dim)] for _ in range(count)],
]
collection.insert(data=data)


collection.flush()
print("insert done")


collection = Collection(name=collection_name)
print(collection.num_entities)

print(collection.describe())
index_params = {
    'metric_type': metric_type,
    'index_type': "IVF_SQ8",
    'params': {'nlist': 1024},
}
collection.create_index(field_name="vector", index_params=index_params)
print("index created")

collection = Collection(name=collection_name)
collection.load()
print("collection loaded")

target_vector = [random.random() for _ in range(dim)]
res = collection.search(data=[target_vector],
                        limit=10,
                        expr="\"int64_field\" > 500 && \"int64_field\" < 1000",
                        anns_field="vector",
                        output_fields=["*"],
                        param={"metric_type":metric_type})
print(f"Search result:\n{res}")
```

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by longjiquan**
fix: #34139

### CODE REVIEW

File: internal/parser/planparserv2/parser_visitor.go [hunk 1]
```diff
@@ -383,18 +383,23 @@ func (v *ParserVisitor) VisitRelational(ctx *parser.RelationalContext) interface
...
 		case parser.PlanParserLE:
-			return LessEqual(leftValue, rightValue)
+			ret = LessEqual(leftValue, rightValue)
 		case parser.PlanParserGT:
-			return Greater(leftValue, rightValue)
+			ret = Greater(leftValue, rightValue)
 		case parser.PlanParserGE:
-			return GreaterEqual(leftValue, rightValue)
+			ret = GreaterEqual(leftValue, rightValue)
 		default:
```
Comment (zhagnlu) [inline internal/parser/planparserv2/parser_visitor.go:396]: what about  EQ and NEQ

Comment (longjiquan) [reply internal/parser/planparserv2/parser_visitor.go:396]: yeah, u r right. thx for the reminder, and i'll fix it.

**Pull request merged**